EmployeeManagementService/EmployeeManagementService.Test/Controller/AdminControllerTest.cs

```diff
@@ -0,0 +1,396 @@
+using EmployeeManagementService.Domain.Services;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EmployeeManagementService.Test.Controller
+{
+    [TestFixture]
+    public class AdminControllerTest
+    {
+        private Mock<IEmployeeService> _employeeService;
+
+        [SetUp]
+        public void Setup()
+        {
+            _employeeService = new Mock<IEmployeeService>();
+        }
+
+        [Test]
+        public async Task GetAllEmployees_Success()
+        {
+            _employeeService.Setup(e => e.GetAllEmployees(It.IsAny<int>(), It.IsAny<int>()))
+                .ReturnsAsync(new List<Domain.Models.Employee>()
+                {
+                    new Domain.Models.Employee()
+                    {
+                        Id = 1,
+                        FirstName = "John",
+                        LastName = "Doe",
+                        Ssn = "123-45-6789",
+                        Username = "jdoe",
+                        Password = new byte[32],
+                        Role = "Employee",
+                        IsLocked = false,
+                        FailedLoginAttempts = 0,
+                        TempPasswordChanged = false,
+                        Status = false,
+                        Active = true
+                    }
+                });
+
+            var controller = new AdminController(_employeeService.Object);
+
+            var response = await controller.GetAllEmployees(1, 10);
+
+            Assert.NotNull(response);
+            Assert.AreEqual(response.GetType(), typeof(OkObjectResult));
+
+            var okObj = (OkObjectResult)response;
+
+            Assert.AreEqual(okObj.StatusCode, 200);
+        }
+
+        [Test]
+        public async Task GetAllEmployees_InternalServerError()
+        {
+            _employeeService.Setup(e => e.GetAllEmployees(It.IsAny<int>(), It.IsAny<int>()))
+                .ThrowsAsync(new Exception());
+
+            var controller = new AdminController(_employeeService.Object);
+
+            var response = await controller.GetAllEmployees(1, 10);
+
+            Assert.NotNull(response);
+            Assert.AreEqual(response.GetType(), typeof(ObjectResult));
+
+            var obj = (ObjectResult)response;
+
+            Assert.AreEqual(obj.StatusCode, 500);
+        }
+
+        [Test]
+        public async Task CreateEmployee_Success()
+        {
+            var newEmployee = new API.DTO.EmployeeDTO()
+            {
+                FirstName = "John",
+                LastName = "Doe",
+                Username = "jdoe",
+                Ssn = "123-45-6789",
+                Password = "T3$T1234",
+                Role = "Employee",
+                Active = true
+            };
+
+            var controller = new AdminController(_employeeService.Object);
+
+            var response = await controller.CreateEmployee(newEmployee);

```

Comment:
how come you did have to mock like the service's create employee method to return success? You did that in reset lock below.